pycbc/inference/models/hierarchical.py

```diff
@@ -599,3 +603,298 @@ def _loglikelihood(self):
             for stat in self.extra_stats_map[lbl]:
                 setattr(self._current_stats, stat, mstats[stat.subname])
         return logl
+
+
+class MultibandRelativeTimeDom(HierarchicalModel):
+    """ Hierarchical heterodyne likelihood for coherent multiband
+    parameter estimation which combines data from space-borne and
+    ground-based GW detectors coherently. Currently, this only
+    supports LISA as the space-borne GW detector.
+
+    Sub models are treated as if the same GW source (such as a GW
+    from stellar-mass BBH) is observed in different frequency bands by
+    space-borne and ground-based GW detectors, then transform all
+    the parameters into the same frame in the sub model level, use
+    `HierarchicalModel` to get the joint likelihood, and marginalize
+    over all the extrinsic parameters supported by `RelativeTimeDom`.
+    Note that LISA submodel only supports the `Relative` for now,
+    for ground-based detectors, please use `RelativeTimeDom`.
+    """
+    name = 'multiband_relative_time_dom'
+
+    def __init__(self, variable_params, submodels, **kwargs):
+        super().__init__(variable_params, submodels, **kwargs)
+
+        # assume the ground-based submodel as the primary model
+        self.primary_model = self.submodels[kwargs['primary_lbl'][0]]
+        self.other_models = self.submodels.copy()
+        self.other_models.pop(kwargs['primary_lbl'][0])
+        self.other_models = list(self.other_models.values())
+
+    def write_metadata(self, fp, group=None):
+        """Adds metadata to the output files
+
+        Parameters
+        ----------
+        fp : pycbc.inference.io.BaseInferenceFile instance
+            The inference file to write to.
+        group : str, optional
+            If provided, the metadata will be written to the attrs specified
+            by group, i.e., to ``fp[group].attrs``. Otherwise, metadata is
+            written to the top-level attrs (``fp.attrs``).
+        """
+        super().write_metadata(fp, group=group)
+        sampattrs = fp.getattrs(group=fp.samples_group)
+        # if a group is specified, prepend the lognl names with it
+        if group is None or group == '/':
+            prefix = ''
+        else:
+            prefix = group.replace('/', '__')
+            if not prefix.endswith('__'):
+                prefix += '__'
+        try:
+            for lbl, model in self.submodels.items():
```

Comment:
@ahnitz I also modified this `write_metadata` method, previously I just copied it from that `MultiSignalModel`, but I noticed there is a main difference between my model and it: doesn't like `MultiSignalModel`, my submodels don't share the same detector data, so I have to save all `lognl`. Is this part correct?

Reply:
@ahnitz Just checked, the numbers saved in the final posterior file makes sense:
![image](https://github.com/gwastro/pycbc/assets/6574526/eabdffba-412f-471b-ba23-90f1c15f4fd4)
![image](https://github.com/gwastro/pycbc/assets/6574526/3f02130a-bd93-45ee-8917-ad0413270278)